### ISSUE TITLE
Liquidsoap 1.4.4 is incompatible with samplerate 0.1.6

### ISSUE THREAD

**Issue by AngusF22** [savonet/liquidsoap#1805]
**Describe the bug**
When trying to install liquidsoap via opam on Debian Stretch, samplerate 0.1.6 produces the following output:
```

[ERROR] The compilation of liquidsoap failed at "/home/user/.opam/opam-init/hooks/sandbox.sh build make".

#=== ERROR while compiling liquidsoap.1.4.4 ===================================#
# context     2.0.7 | linux/x86_64 | ocaml-base-compiler.4.08.0 | https://opam.ocaml.org#9018b2a4
# path        ~/.opam/4.08.0/.opam-switch/build/liquidsoap.1.4.4
# command     ~/.opam/opam-init/hooks/sandbox.sh build make
# exit-code   2
# env-file    ~/.opam/log/liquidsoap-1559-c61e7b.env
# output-file ~/.opam/log/liquidsoap-1559-c61e7b.out
### output ###
# [...]
# /home/user/.opam/4.08.0/.opam-switch/build/samplerate.0.1.6/_build/default/src/samplerate_stubs.c:210: undefined reference to `src_get_channels'
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking
# ../Makefile.rules:107: recipe for target 'liquidsoap' failed
# make[2]: *** [liquidsoap] Error 2
# make[2]: Leaving directory '/home/user/.opam/4.08.0/.opam-switch/build/liquidsoap.1.4.4/src'
# ../Makefile.rules:69: recipe for target 'all-auto-ocaml-prog' failed
# make[1]: *** [all-auto-ocaml-prog] Error 2
# make[1]: Leaving directory '/home/user/.opam/4.08.0/.opam-switch/build/liquidsoap.1.4.4/src'
# Makefile.rules:29: recipe for target 'all-subdirs' failed
# make: *** [all-subdirs] Error 2


```
**To Reproduce**
Follow the opam install instructions to the letter.

**Expected behavior**
I expected it to install correctly.

**Version details**
Linux openmediavault 4.19.0-0.bpo.9-amd64 #1 SMP Debian 4.19.118-2+deb10u1~bpo9+1 (2020-06-09) x86_64 GNU/Linux
opam 2.0.7

**Install method**
Installed via opam.

**Solution**
I was able to install liquidsoap by downgrading to samplerate 0.1.4 via the following command:
opam install samplerate=0.1.4

**Comment by toots** [comment]
Hi,

The issue comes from your version of the underlying `libsample` library. Downgrading is indeed the fix. 

Closing here since there isn't much we can do at the moment. I will have a look to see if we can avoid using this API.

**Comment by toots** [comment]
It looks like the missing API, `src_get_channels` has been released since `2016` with version `0.1.9` of the upstream library. Your system is based on debian, which is slow to catch-up on recent version but `0.1.9` has been available at least since the last stable, `buster`, released on July of 2019. Unfortunately, it seems pretty burdensome for use to keep targeting a version of the library that was first released 5 years ago. Hopefully you find a way to upgrade your version as well in the near future. Meanwhile, you can pin the older version of `ocaml-samplerate`: `opam install samplerate.0.1.3`.
